src/commands.ts

```diff
@@ -61,7 +61,10 @@ export class Commands<C extends Context> {
     private _languages: Set<LanguageCode | "default"> = new Set();
     private _scopes: Map<string, Array<Command<C>>> = new Map();
     private _commands: Command<C>[] = [];
+
+    private _dirty: boolean = false;
```

Comment:
I'd like a more descriptive name, like the verbose `_isComposerPopulates` or `_isComposerDirty`. WDYT?